supervisor/process.go

```diff
@@ -0,0 +1,411 @@
+package supervisor
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/pkg/errors"
+	"os/exec"
+	"strings"
+	"sync"
+	"syscall"
+	"time"
+)
+
+const (
+	DefaultLineBufferSize = 16384
+	DefaultStreamChanSize = 200
+)
+
+type ProcessState uint8
+
+const (
+	Initial ProcessState = iota
+	Starting
+	Running
+	Stopping
+	Finished
+	Interrupted
+	Fatal
+)
+
+func (s ProcessState) String() string {
+	switch s {
+	case Initial:
+		return "initial"
+	case Starting:
+		return "starting"
+	case Running:
+		return "running"
+	case Stopping:
+		return "stopping"
+	case Finished:
+		return "finished"
+	case Fatal:
+		return "fatal"
+	default:
```

Comment:
Feels like Interrupted is missing from here.

Reply:
Thanks. This has been fixed